### ISSUE TITLE
[BUG] og_image templatetag with amazon s3

### ISSUE THREAD

**Issue by parleweb** [coderedcorp/coderedcms#240]
I have aws s3 as static files.
In that case, the generate url for image is wrong : have django host url + amazon static url (two different server). I made a quick fix with a site_url if settings.STATIC_URL start by 'http' in og_image template tags and its working. As follow : 

```
@register.simple_tag(takes_context=True)
def aws_og_image(context, page): # duplicate template tag for amazon
    if settings.STATIC_URL[:4].lower() == 'http':
        site_url = ''
    else:
        site_url = context['request'].site.root_url
```
Also, this is not working for tweeter as there is no {% block og_image %} and i don't want to modify original base.html.

**Comment by vsalvino** [comment]
Good catch! I think this is a bug in the way we generate the absolute path to the static string. It is a bit too "hard-coded" right now.

**Comment by wamcvey** [comment]
I'm also hosting my og:image on a CDN and am running into this issue. I did a quick workaround to hardcode my og:image URL into a custom defined og_image block for my app and another one for my site. However, this is not only kludgy, it's only even a partial "fix" because while there is an og_image block that you can override for the facebook image, there is no block surrounding the broken og_image tag when it is used for twitter:image

**Comment by vsalvino** [comment]
If anyone using s3 or other object storage can help contribute a fix, it would be much appreciated.

1. The missing `{% block og_image %}` should definitely be added to the twitter card, and anywhere else it is used (all metadata should have a similar block for easy overriding).

2. The code that outputs the image URL is here: https://github.com/coderedcorp/coderedcms/blob/67871b2bd871d6f76613f297666a468894bd6a4b/coderedcms/templatetags/coderedcms_tags.py#L46

The only requirement is that the URL is an absolute URL. Relative URLs such as `/static/img.jpg` will cause problems with some opengraph parsers.

**Comment by parleweb** [comment]
Changing line 47 

`site_url = context['request'].site.root_url`

by 

```
    if settings.STATIC_URL[:4].lower() == 'http':
        site_url = ''
    else:
        site_url = context['request'].site.root_url
```
just works fine for me, as static cloud always come with a https://cloud_service_server/...
for example, setting up Amazon S3 add this kind of new STATIC_URL definition :

```
AWS_STORAGE_BUCKET_NAME = 'mycloudsubdomain'
AWS_LOCATION = 'mystaticfilesdir'
AWS_S3_CUSTOM_DOMAIN = '%s.s3.amazonaws.com' % AWS_STORAGE_BUCKET_NAME
STATIC_URL = 'https://%s/%s/' % (AWS_S3_CUSTOM_DOMAIN, AWS_LOCATION)

```

**Comment by bahoo** [comment]
One note, this needs to look at `MEDIA_URL` instead of `STATIC_URL` since those OG images are uploaded into the admin, and served as media. ( Though of course it's very often the case that those two go hand-in-hand. )

I submitted a PR with this change: https://github.com/coderedcorp/coderedcms/pull/257

**Comment by vsalvino** [comment]
Fixed in 0.16.3
